yjit_codegen.c

```diff
@@ -3445,6 +3445,28 @@ gen_getglobal(jitstate_t* jit, ctx_t* ctx)
     return YJIT_KEEP_COMPILING;
 }
 
+static codegen_status_t
+gen_setglobal(jitstate_t* jit, ctx_t* ctx)
+{
+    ID gid = jit_get_arg(jit, 0);
+
+    // Save YJIT registers
+    yjit_save_regs(cb);
+
+    mov(cb, C_ARG_REGS[0], imm_opnd(gid));
+
+    x86opnd_t val = ctx_stack_pop(ctx, 1);
+
+    mov(cb, REG0, val);
```

Comment:
I think you probably want to move `val` into `C_ARG_REGS[1]`

Reply:
Whoops, fixed that and the one below.